.ibm/pipelines/cluster/osd-gcp/create-osd.sh

```diff
@@ -59,22 +64,19 @@ echo "    api_url: $(ocm describe cluster $CLUSTER_ID --json | jq -rc '.api.url'
 echo "    $KUBEADMIN_USER: $KUBEADMIN_PASSWORD" >> $CLUSTER_CONFIG
 echo "    cli: oc login $(ocm describe cluster $CLUSTER_ID --json | jq -rc '.api.url') --username $KUBEADMIN_USER --password $KUBEADMIN_PASSWORD" >> $CLUSTER_CONFIG
 
-echo
-cat $CLUSTER_CONFIG
-echo
-
 export KUBECONFIG=$WORKSPACE/kubeconfig
 rm -rvf $KUBECONFIG
-retries=50
-until [[ $retries == 0 ]]; do
-    echo "Attempting to login to get kubeconfig - $retries attempts remaining..."
-    oc login $(ocm describe cluster $CLUSTER_ID --json | jq -rc '.api.url') --username $KUBEADMIN_USER --password $KUBEADMIN_PASSWORD >/dev/null 2>&1 && break
+CLUSTER_API_URL=$(ocm describe cluster $CLUSTER_ID --json | jq -rc '.api.url')
+for i in {1..50}; do
+    echo "Attempt $i: Logging in..."
+    if oc login "$CLUSTER_API_URL" --username "$KUBEADMIN_USER" --password "$KUBEADMIN_PASSWORD" --insecure-skip-tls-verify=true; then
+        echo "Login successful!"
+        ocm describe cluster $CLUSTER_ID > $WORKSPACE/cluster-info.yaml
+        exit 0
+    fi
+    echo "Login failed. Retrying in 30 seconds..."
     sleep 30
```

Comment:
If all login attempts fail, the script now explicitly returns an error (exit 1) instead of failing silently.

for i in {1..50}; do
    echo "Attempt $i: Logging in..."
    
    if oc login "$CLUSTER_API_URL" --username "$KUBEADMIN_USER" --password "$KUBEADMIN_PASSWORD" --insecure-skip-tls-verify=true; then
        echo "Login successful!"
        ocm describe cluster "$CLUSTER_ID" > "$WORKSPACE/cluster-info.yaml"
        exit 0
    fi

    echo "Login failed. Retrying in 30 seconds..."
    sleep 30
done

echo "Error: All login attempts failed."
exit 1